### ISSUE TITLE
Improved mixing of back translation in drafting configuration

### ISSUE THREAD

**Issue by mmartin9684-sil** [sillsdev/silnlp#837]
During preprocessing, the current mixed source logic does a random 50/50 split of the main reference translation and the back translation to create the source side of the training dataset.  That is, 50% of the source verses are taken from the main reference translation, and the other 50% of the verses are taken from the back translation.  This is not likely to be the best way to get value out of the back translation project.

Most of the time, the back translation will be the most valuable for verses where the target verse is least similar to the source verse.  This lack of similarity is likely because the project has needed to explicate the verse in more detail or express the idea very differently for various reasons.  For these verses, the project's back translation of the verse is likely to be a better representation of the project's verse translation.  Pairing the back translation text as the source text for this verse is likely to be a better training setup.  Altering the mixed source logic to have, e.g., an alignment-based score threshold, and only using the BT verse when the score threshold is not met, would ensure that the back translation text is only used when it's most likely to be helpful.

I believe that the mixed source logic is applied to the train, validation, and test sets.  If that's the case, another improvement would be to limit the mixing to just the training set.  There's no need to mix the BT into the validation and test sets.  When the model is actually used to create a draft, it will only be used on the reference translation, and the best validation / test set would consist of only the reference translation verses.
